tests/test_qgis_bot.py

```diff
@@ -17,24 +17,92 @@
 #  along with pytest-qgis.  If not, see <https://www.gnu.org/licenses/>.
 #
 import pytest
-from qgis.core import QgsGeometry
+from qgis.core import QgsFieldConstraints, QgsGeometry
 from qgis.gui import QgsAttributeDialog
 
 
-@pytest.mark.parametrize("bot", ["qgis_bot", "module_qgis_bot"])
-def test_create_feature_with_attribute_dialog(layer_points, bot, request):
-    bot = request.getfixturevalue(bot)
+def test_create_simple_feature_with_attribute_dialog(layer_points, qgis_bot):
     layer = layer_points
     count = layer.featureCount()
 
     layer.startEditing()
-    feat = bot.create_feature_with_attribute_dialog(
+    feat = qgis_bot.create_feature_with_attribute_dialog(
         layer, QgsGeometry.fromWkt("POINT(0,0)")
     )
     assert layer.featureCount() == count + 1
     assert feat["bool_field"] is False  # With normal way of creating, it would be NULL.
 
 
+@pytest.mark.parametrize("ignore_warnings", [False, True])
```

Comment:
I think tests should test one thing only and actual test statement should not depend on parameter values. Imho these should be separated to two or actually to three different tests.

Something like:
```python
@pytest.fixture()
def layer_with_soft_constraint(layer_points):
    """setup the layer"""
    # Set not-null constraint with SOFT strength
    fields = layer_points.fields()
    field_idx = fields.indexOf("text_field")
    layer_points.setFieldConstraint(
        field_idx,
        QgsFieldConstraints.Constraint.ConstraintNotNull,
        QgsFieldConstraints.ConstraintStrengthSoft,
    )

    layer_points.startEditing()

    return layer_points


class TestFeatureCreationWhenNotRaisingFromSoftBreaks:
    @pytest.fixture(scope="class")
    def layer(layer_with_soft_constraint):
      self.count_before = layer_with_soft_constraint.featureCount()
      return layer_with_soft_constraint

    @pytest.fixture(scope="class)
    def feature(self, point_layer):
        self.feature_count_before = layer.featureCount()
        return qgis_bot.create_feature_with_attribute_dialog(
            layer, QgsGeometry.fromWkt("POINT(0,0)"), raise_for_warnings=False
        )

    def test_feature_gets_created(self, layer, feature):
        assert layer.featureCount() == self.count_before  + 1

    def test_default_values_should_be_assigned(self, feature):
        feat = qgis_bot.create_feature_with_attribute_dialog(
            layer, QgsGeometry.fromWkt("POINT(0,0)"), raise_for_warnings=False
        )
        assert feat["bool_field"] is False

def test_should_raise_valueerror_on_soft_constraint_break_when_asked(layer):
    with pytest.raises(ValueError, match="value is NULL"):
        qgis_bot.create_feature_with_attribute_dialog(
            layer,
            QgsGeometry.fromWkt("POINT(0,0)"),
            raise_from_warnings=True,
        )
```

Reply:
I'll refactor the tests.